lib/models/service.ts

```diff
@@ -54,53 +49,56 @@ export interface INewService
  * Existing Service records have a version number.
  */
 export interface IRetrievedService
-  extends ISerializableService,
+  extends IService,
     DocumentDb.RetrievedDocument,
     IVersionedModel {
   readonly id: NonEmptyString;
   readonly kind: "IRetrievedService";
 }
 
-export interface IAuthorizedService extends IBaseService {
-  // list of authorized fiscal codes
-  readonly authorizedRecipients?: ReadonlySet<FiscalCode>;
-}
-
-export function toAuthorizedService(
-  service: ISerializableService
-): IAuthorizedService {
-  return {
-    ...service,
-    authorizedRecipients: service.authorizedRecipients
-      ? new Set(service.authorizedRecipients.filter(isFiscalCode))
-      : new Set()
-  };
-}
-
-export function toSerializableService(
-  service: IAuthorizedService
-): ISerializableService {
-  return {
-    ...service,
-    authorizedRecipients: service.authorizedRecipients
-      ? Array.from(service.authorizedRecipients).filter(isFiscalCode)
-      : []
-  };
+/**
+ * Converts an Array or a Set of strings to a ReadonlySet of fiscalCodes.
+ * 
+ * We need to handle Arrays as this method is called by database finders
+ * who retrieve a plain json object.
+ * 
+ * We need to handle Sets as this method is called on IService objects
+ * passed to create(IService) and update(IService) model methods.
+ * 
+ * @param authorizedRecipients 
+ */
+export function toAuthorizedRecipientsSet(
```

Comment:
This is a little ugly, the method accepts the union of two types but it implements two different logics for the two types (so what's the advantage of having a single method accepting an union type?)

Having two separate methods would make it explicit that the two sources are handled differently.